runtime-light/stdlib/string/regex-functions.cpp

```diff
@@ -543,8 +574,17 @@ Optional<int64_t> f$preg_match_all(const string &pattern, const string &subject,
   return entire_match_count;
 }
 
-Optional<string> f$preg_replace(const string &pattern, const string &replacement, const string &subject, int64_t limit, int64_t &count) noexcept {
-  count = 0;
+Optional<string> f$preg_replace(const string &pattern, const string &replacement, const string &subject, int64_t limit,
+                                Optional<std::variant<std::monostate, std::reference_wrapper<int64_t>>> opt_count) noexcept {
```

Comment:
NIT: alias for `Optional<std::variant<std::monostate, std::reference_wrapper<T>>>`. It looks awfully large

Reply:
Yeah, it looks awful. But I think we should not have an alias for that. Having such alias would hide the complexity of this type, e.g. the variant semantics under the hood. Ideally, I'd create a wrapper type with its own API since such a pattern is gonna be used many time across runtime. But let's have it as a separate task